src/cmd/ksh93/sh/init.c

```diff
@@ -1631,30 +1630,22 @@ static_fn Namval_t *next_svar(Namval_t *np, Dt_t *root, Namfun_t *fp) {
 
 static_fn Namval_t *create_svar(Namval_t *np, const void *vp, int flag, Namfun_t *fp) {
     const char *name = vp;
-    struct Svars *sp = (struct Svars *)fp;
-    const char *cp = name;
-    int i = 0, n;
-    Namval_t *nq = NULL;
+    Svars_t *sp = (Svars_t *)fp;
     Shell_t *shp = sp->sh;
 
-    if (!name) return (sp->parent);
-    while ((i = *cp++) && i != '=' && i != '+' && i != '[') {
-        ;  // empty loop
-    }
-    n = (cp - 1) - name;
-    for (i = 0; i < sp->numnodes; i++) {
-        nq = nv_namptr(sp->nodes, i);
-        if ((n == 0 || strncmp(name, nq->nvname, n) == 0) && nq->nvname[n] == 0) goto found;
-    }
-    nq = NULL;
-found:
-    if (nq) {
-        fp->last = (char *)&name[n];
-        shp->last_table = sp->parent;
-    } else {
-        errormsg(SH_DICT, ERROR_exit(1), e_notelem, n, name, nv_name(np));
+    assert(name);
+    for (int i = 0; i < sp->numnodes; i++) {
+        Namval_t *nq = nv_namptr(sp->nodes, i);
+        if (strcmp(name, nq->nvname) == 0) {
```

Comment:
What if name contains an assignment ? I have seen pieces of code where assignments like `foo=bar` are passed as a single name parameter.

Reply:
That isn't an issue for this function. The sole use of this function is to assign values to the `.sh.sig.*` attributes. It's not used in any context where your scenario can happen. I'll add a doc-string comment at the top of the function.